spec/listeners/automation_rule_listener_spec.rb

```diff
@@ -146,6 +152,117 @@
     end
   end
 
+  describe '#conversation_updated with contacts attributes' do
+    before do
+      Contact.first.update!(custom_attributes: { customer_type: 'platinum', signed_in_at: '2022-01-19' },
+                            additional_attributes: { 'company': 'Marvel' })
+      Contact.last.update!(custom_attributes: { customer_type: 'platinum', signed_in_at: '2022-01-19' },
+                           additional_attributes: { 'company': 'Marvel' })
+
+      automation_rule.update!(
+        event_name: 'conversation_updated',
+        name: 'Call actions conversation updated',
+        description: 'Add labels, assign team after conversation updated',
+        conditions: [
+          {
+            attribute_key: 'company',
+            filter_operator: 'equal_to',
+            values: ['Marvel'],
+            query_operator: 'AND'
+          }.with_indifferent_access,
+          {
+            attribute_key: 'customer_type',
+            filter_operator: 'equal_to',
+            values: ['platinum'],
+            query_operator: nil
+          }.with_indifferent_access
+        ]
+      )
+    end
+
+    let!(:event) do
+      Events::Base.new('conversation_updated', Time.zone.now, { conversation: conversation })
+    end
+
+    context 'when rule matches with additional_attributes and custom_attributes' do
+      it 'triggers automation rule to assign team' do
+        expect(conversation.team_id).not_to eq(team.id)
+
+        automation_rule
+        listener.conversation_updated(event)
+
+        conversation.reload
+        expect(conversation.team_id).to eq(team.id)
+      end
+
+      it 'triggers automation rule to add label and assign best agents' do
+        expect(conversation.labels).to eq([])
+        expect(conversation.assignee).to be_nil
+
+        automation_rule
+        listener.conversation_updated(event)
+
+        conversation.reload
+        expect(conversation.labels.pluck(:name)).to eq(%w[support priority_customer])
```

Comment:
let's use `contain_exactly` for array going forward is order is not important
ref: https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/contain-exactly-matcher